scripts/run-cppcheck.sh

```diff
@@ -22,6 +22,9 @@ mv cppcheck_tmp.txt cppcheck.txt
 grep -v "'AmiVectorArray' does not have a operator=" cppcheck.txt > cppcheck_tmp.txt
 mv cppcheck_tmp.txt cppcheck.txt
 
+grep -v "Obsolete function 'alloca' called." cppcheck.txt > cppcheck_tmp.txt
```

Comment:
Shall we just remove the `alloca`s?

Reply:
Alternatively use `--exitcode-suppressions=` (or less preferred, `--suppress=`), which is cleaner and probably more stable

Reply:
@FFroehlich Do we have any evidence that using std::vector will have a noticeable performance impact? Otherwise I'd just change it to that. With clang, g++, intel compilers we could use variable length arrays, but this will cause trouble with msvc++...

Reply:
I believe there was a noticable performance impact when considering second order forward sensitivities for the Swameye model.

Reply:
Then we stay with `alloca`, I guess